frontend/src/App.js

```diff
@@ -0,0 +1,58 @@
+import React, { Component } from 'react';
+import ChannelList from './Components/ChannelList';
+import MessageList from './Components/MessageList';
+
+class App extends Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			__active_channel: null,
+			data: {},
+		};
+		this.ws = new WebSocket('ws://localhost:8000/ws/');
+		this.ws.onerror = e => console.log(e);
+		this.ws.onopen = e => console.log('WS connnected');
+		this.ws.onmessage = this.handleIncomingData;
+	}
+
+	handleIncomingData = payload => {
+		const data = JSON.parse(payload.data);
+		for (let channel in data) {
+			if (this.state[channel] === undefined) {
+				this.setState(state => {
+					state.data[channel] = data[channel];
+					return state;
+				});
+			}
+			if (this.state[channel] !== undefined) {
+				this.setState(state => {
+					state.data[channel].push(...data[channel]);
+					return state;
+				});
+			}
+		}
+	};
+
+	changeActiveChannel = channel_name => {
+		this.setState({ __active_channel: channel_name });
+	};
+
+	render() {
+		return (
+			<div className='App'>
+				<ChannelList
+					active_channel={this.state.__active_channel}
```

Comment:
why add the `__` to active channel
`__` makes me think its a private field, so doesnt make sense to pass it down

Reply:
the list that __activechannel is apart of is passed from the DB to th front end. I didn't want to deal with some edge case where someone named their channel 'active_channel'

Reply:
err lets go over this on tuesday in person